### PR TITLE
refactor: billing endpoints

### PR DESCRIPTION
**Pull request by ArmanNik**

### CODE REVIEW

File: src/lib/stores/billing.ts [hunk 1]
```diff
@@ -211,7 +211,10 @@ export async function checkPaymentAuthorizationRequired(org: Organization) {
 
 export async function paymentExpired(org: Organization) {
     if (!org?.paymentMethodId) return;
-    const payment = await sdk.forConsole.billing.getPaymentMethod(org.paymentMethodId);
+    const payment = await sdk.forConsole.billing.getOrganizationPaymentMethod(
```
Comment (lohanidamodar) [inline src/lib/stores/billing.ts:214]: Where is this method used?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Left a query. Also where are the similar changes for billingAddresses?

File: src/lib/stores/billing.ts (hunk 1 above)
Comment (ArmanNik) [reply src/lib/stores/billing.ts:214]: We use this on load to check if there are any expired payments

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
We need similar changes applied for `billingAddress` as well. I don't see those.

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Hey while loading billing address in the organizations billing section, we should always use the endpoint from organization instead from account
image

I don't yet see that change?

**Comment by ArmanNik** [comment]
@lohanidamodar there's no change because we only fetch all the addresses never only one. 

If there's also a new endpoint for listing addresses, I wasn't informed and it's not mentioned in the linear task either :thinking:

**Comment by lohanidamodar** [comment]
> @lohanidamodar there's no change because we only fetch all the addresses never only one.
> 
> If there's also a new endpoint for listing addresses, I wasn't informed and it's not mentioned in the linear task either 🤔

How do you display the Organization's assigned billing address? don't you need to fetch the billing address by ID when people are looking at the billing tab for the organization?

**Comment by lohanidamodar** [comment]
> > > @lohanidamodar there's no change because we only fetch all the addresses never only one.
> > > If there's also a new endpoint for listing addresses, I wasn't informed and it's not mentioned in the linear task either 🤔
> > 
> > 
> > How do you display the Organization's assigned billing address? don't you need to fetch the billing address by ID when people are looking at the billing tab for the organization?
> 
> We are already fetching the addressList when we load the page so I'm just finding it among them
> 
> ```
> $addressList?.billingAddresses?.find(
>         (address) => address.$id === $organization?.billingAddressId
>     );
> ```

As we discussed, lets use the new endpoint through `/organizations/billing-addresses/:billingAddressId` to fetch the organization's billing address.

**Review by lohanidamodar — COMMENTED** [review]
Looks good. However I think this should target `cloud` branch.

**Review by lohanidamodar — APPROVED** [review]

**Pull request merged**